ledger/participant-integration-api/src/main/scala/platform/apiserver/error/RejectionGenerator.scala

```diff
@@ -0,0 +1,147 @@
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case e: LfInterpretationError.ContractIdInContractKey =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case LfInterpretationError.ValueExceedsMaxNesting =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case e: LfInterpretationError.ContractIdComparability =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+      }
+    }
+
+    def processInterpretationError(err: LfError.Interpretation.Error, detailMessage: Option[String]): BaseError =
+      err match {
+        case Interpretation.Internal(location, message) =>
+          LedgerApiErrors.InternalError.Interpretation(location, message, detailMessage)
+        case m @ Interpretation.DamlException(error) => processDamlException(error, m.message)
+      }
+
+    def processLfError(error: LfError) = {
+      val transformed = error match {
+        case LfError.Package(packageError)          => processPackageError(packageError)
+        case LfError.Preprocessing(processingError) => processPreprocessingError(processingError)
+        case LfError.Interpretation(interpretationError, detailMessage) =>
+          processInterpretationError(interpretationError, detailMessage)
+        case LfError.Validation(validationError) => processValidationError(validationError)
+        case e
+          if e.message.contains("requires authorizers") => // Keeping this around as a string match as daml is not yet generating LfError.InterpreterErrors.Validation
+          LedgerApiErrors.InterpreterErrors.AuthorizationError.Reject(e.message)
+      }
+      toGrpc(transformed)
+    }
+
+    val rej = cause match {
+      case ErrorCauseExport.DamlLf(error) => processLfError(error)
+      case x: ErrorCauseExport.LedgerTime =>
+        toGrpc(LedgerApiErrors.CommandPreparation.FailedToDetermineLedgerTime.Reject(x.explain))
+    }
+    Some(rej)
+  }
+
+  override def submissionResult(commands: domain.Commands,
+                                result: Try[state.v2.SubmissionResult]): Option[Try[Unit]] = {
+    result match {
+      case Success(state.v2.SubmissionResult.Acknowledged)                    => None
+      case Success(grpcError @ state.v2.SubmissionResult.SynchronousError(_)) => Some(Failure(grpcError.exception))
+      case Failure(_)                                                         => None
+    }
+  }
+
+  override def validationFailure(reject: StatusRuntimeException): StatusRuntimeException = {
+    implicit val errorContext: ErrorLoggingContext = this.loggingContext(TraceContext.empty)
+    val description                                = reject.getStatus.getDescription
+    reject.getStatus.getCode match {
+      case Code.INVALID_ARGUMENT =>
+        if (description.startsWith("Missing field:")) {
```

Comment:
Should these prefixes become constants? (I'm guessing the same prefixes are used somewhere in the codebase to create these descriptions)